render_blender.py

```diff
@@ -26,149 +30,209 @@
 bpy.data.objects['Sun'].rotation_euler[0] += 180
 
-
-def parent_obj_to_camera(b_camera):
-    origin = (0, 0, 0)
-    b_empty = bpy.data.objects.new("Empty", None)
-    b_empty.location = origin
-    b_camera.parent = b_empty  # setup parenting
-
-    scn = bpy.context.scene
-    scn.objects.link(b_empty)
-    scn.objects.active = b_empty
-    return b_empty
-
-
-scene = bpy.context.scene
-scene.render.resolution_x = 600
-scene.render.resolution_y = 600
-scene.render.resolution_percentage = 100
-scene.render.alpha_mode = 'TRANSPARENT'
+# Place camera
 cam = scene.objects['Camera']
 cam.location = (0, 1, 0.6)
+cam.data.lens = 35
+cam.data.sensor_width = 32
+
 cam_constraint = cam.constraints.new(type='TRACK_TO')
 cam_constraint.track_axis = 'TRACK_NEGATIVE_Z'
 cam_constraint.up_axis = 'UP_Y'
-b_empty = parent_obj_to_camera(cam)
-cam_constraint.target = b_empty
 
-model_identifier = os.path.split(os.path.split(args.obj)[0])[1]
-fp = os.path.join(args.output_folder, model_identifier, model_identifier)
-scene.render.image_settings.file_format = 'PNG'  # set output format to .png
+cam_empty = bpy.data.objects.new("Empty", None)
+cam_empty.location = (0, 0, 0)
+cam.parent = cam_empty
 
-from math import radians
+scene.collection.objects.link(cam_empty)
+context.view_layer.objects.active = cam_empty
+cam_constraint.target = cam_empty
 
 stepsize = 360.0 / args.views
 rotation_mode = 'XYZ'
 
-for output_node in [depth_file_output, normal_file_output, albedo_file_output]:
-    output_node.base_path = ''
+model_identifier = os.path.split(os.path.split(args.obj)[0])[1]
+fp = os.path.join(os.path.abspath(args.output_folder), model_identifier, model_identifier)
 
 for i in range(0, args.views):
-    print("Rotation {}, {}".format((stepsize * i), radians(stepsize * i)))
+    print("Rotation {}, {}".format((stepsize * i), math.radians(stepsize * i)))
 
-    scene.render.filepath = fp + '_r_{0:03d}'.format(int(i * stepsize))
-    depth_file_output.file_slots[0].path = scene.render.filepath + "_depth.png"
-    normal_file_output.file_slots[0].path = scene.render.filepath + "_normal.png"
-    albedo_file_output.file_slots[0].path = scene.render.filepath + "_albedo.png"
+    render_file_path = fp + '_r_{0:03d}'.format(int(i * stepsize))
+
+    scene.render.filepath = render_file_path
+    depth_file_output.file_slots[0].path = render_file_path + "_depth"
+    normal_file_output.file_slots[0].path = render_file_path + "_normal"
+    albedo_file_output.file_slots[0].path = render_file_path + "_albedo"
+    id_file_output.file_slots[0].path = render_file_path + "_id"
 
     bpy.ops.render.render(write_still=True)  # render still
 
-    b_empty.rotation_euler[2] += radians(stepsize)
+    # remove frame number from file names
+    for n in glob(render_file_path+'_*'):
+        os.rename(n, re.sub(r'(\d+)\.([^\.]+)$', r'.\2', n))
+
+    cam_empty.rotation_euler[2] += math.radians(stepsize)
+
+#bpy.ops.wm.save_as_mainfile(filepath='debug.blend')
```

Comment:
Remove

Reply:
As I mentioned, I think most of the users adjust the script to there needs. Users that have an background in engineering or whatever, but are often not that familiar with blender...

Reply:
Ok, then comment what this line does.